lightyear/src/client/prediction/prespawn.rs

```diff
@@ -177,6 +178,10 @@ impl PreSpawnedPlayerObjectPlugin {
                                 trace!(?kind, "using kind for hash");
                                 kind.hash(&mut hasher)
                             });
+                            // if a user salt is provided, hash after the sorted component list
```

Comment:
The salt is to avoid hash collisions between multiple entities?

Reply:
if two players fire on the same tick, the hashes would otherwise collide because the bullets would have identical archetypes and tick.